File: findDisappearedNumbers.java
//TC:O(n)
class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        List<Integer> result = new ArrayList<>();
        if(nums == null || nums.length == 0) return result;
        
        int n = nums.length;
        for(int i=0; i<n;i++){
            int index = Math.abs(nums[i]) - 1;
            if(nums[index] > 0){
                nums[index]*= -1;
            }
        }
        
        for(int i=0; i < n; i++){
            if(nums[i] > 0) result.add(i+1);
            else nums[i]*= -1;
        }
        return result;
    }
}

File: findDisappearedNumbers.js
// Time Complexity : O(n) where n is nums length
// Space Complexity : O(1)
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this : No

/**
 * @param {number[]} nums
 * @return {number[]}
 */
 var findDisappearedNumbers = function(nums) {
    let result = [];
    for(let i=0; i< nums.length; i++) {
        let index = Math.abs(nums[i]) - 1;
        if(nums[index] > 0) nums[index] *= -1;
    }
    for(let i=0; i< nums.length; i++) {
        if(nums[i] > 0) result.push(i+1);
        else nums[i] *= -1;
    }
    return result;
};